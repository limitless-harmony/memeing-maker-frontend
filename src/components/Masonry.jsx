import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { calculateRem } from 'styles';

const Masonry = ({ children }) => {
  return <StyledMasonry>{children}</StyledMasonry>;
};

const StyledMasonry = styled.div`
  column-count: 2;
  column-gap: ${calculateRem(11)};
  column-fill: balance;
  width: 100%;
  margin: 0 0 ${calculateRem(60)};
`;

Masonry.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
};

export default Masonry;
