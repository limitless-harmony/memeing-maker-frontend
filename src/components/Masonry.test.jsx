import React from 'react';
import { shallow } from 'enzyme';
import Masonry from 'components/Masonry';

describe('Testing the Masonry component', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(
      <Masonry>
        <div>Anything can be here</div>
        <div>Anything can be here</div>
      </Masonry>
    );
    expect(wrapper.find('div'));
    expect(wrapper).toMatchSnapshot();
  });
});
