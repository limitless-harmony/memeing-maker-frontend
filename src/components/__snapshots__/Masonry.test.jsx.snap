// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Masonry component renders without crashing 1`] = `
<styled.div>
  <div>
    Anything can be here
  </div>
  <div>
    Anything can be here
  </div>
</styled.div>
`;
